source/common/http/conn_manager_impl.h

```diff
@@ -452,6 +452,10 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
   TimeSource& time_source_;
   bool remote_close_{};
   bool enable_internal_redirects_with_body_{};
+
+  // The number of requests accumulated on the current connection.
+  uint64_t accumulated_requests_{};
+  bool requests_overflow_{false};
```

Comment:
delete `requests_overflow_`. Just compare `accumulated_requests_` to the limit when needed.